Incorrect parsing results for toml multi-line strings in Windows

**Describe the bug**

It seems that CRLF is not handled correctly; toml is not parsed correctly in Windows.

Toml allows CRLF for newline.
https://toml.io/en/v1.0.0#spec

**Steps to Reproduce**

1. Create a script for test.

`toml_test.ts`
```typescript
import { parse } from "https://deno.land/std@0.208.0/toml/mod.ts";
import { assertEquals } from "https://deno.land/std@0.208.0/testing/asserts.ts";

Deno.test({
  name: "toml parse",
  fn: () => {
    const raw = `[data]\r\nfoo = """\r\nfoo\r\nbar\\\r\n"""`
    const parsed = parse(raw);
    assertEquals(parsed, { data: { foo: "foo\r\nbar" } });
  },
});
```

This is a parsed toml string.
```toml
[data]
foo = """
foo
bar\
"""
```

2. `deno test -A toml_test.ts`
3. Errors
```
toml parse => ./toml_test.ts:4:6
error: AssertionError: Values are not equal.


    [Diff] Actual / Expected


    {
      data: {
-       foo: "\r\nfoo\r\nbar\n\n",
+       foo: "foo\r\nbar",
      },
    }

  throw new AssertionError(message);
        ^
    at assertEquals (https://deno.land/std@0.208.0/assert/assert_equals.ts:53:9)
    at fn (file:///C:/Users/uga/AppData/Local/nvim/toml_test.ts:9:5)
```

**Expected behavior**

See above test.

Here is the definition of a toml string.
https://toml.io/en/v1.0.0#string

**Environment**

<!-- please complete the following information -->

- OS: Windows 11
- deno version:
```
deno 1.38.5 (release, x86_64-pc-windows-msvc)
v8 12.0.267.1
typescript 5.2.2
```
- std version: 0.208

--- a/toml/_parser.ts
+++ b/toml/_parser.ts
@@ -481,13 +481,20 @@ export function MultilineBasicString(
     return failure();
   }
   if (scanner.char() === "\n") {
-    // The first newline is trimmed
+    // The first newline (LF) is trimmed
     scanner.next();
+  } else if (scanner.slice(0, 2) === "\r\n") {
+    // The first newline (CRLF) is trimmed
+    scanner.next(2);
   }
   const acc: string[] = [];
   while (scanner.slice(0, 3) !== '"""' && !scanner.eof()) {
     // line ending backslash
     if (scanner.slice(0, 2) === "\\\n") {
+      scanner.next();
+      scanner.nextUntilChar({ comment: false });
+      continue;
+    } else if (scanner.slice(0, 3) === "\\\r\n") {
       scanner.next();
       scanner.nextUntilChar({ comment: false });
       continue;
@@ -525,8 +532,11 @@ export function MultilineLiteralString(
     return failure();
   }
   if (scanner.char() === "\n") {
-    // The first newline is trimmed
+    // The first newline (LF) is trimmed
     scanner.next();
+  } else if (scanner.slice(0, 2) === "\r\n") {
+    // The first newline (CRLF) is trimmed
+    scanner.next(2);
   }
   const acc: string[] = [];
   while (scanner.slice(0, 3) !== "'''" && !scanner.eof()) {

--- a/toml/parse_test.ts
+++ b/toml/parse_test.ts
@@ -100,6 +100,27 @@ Violets are\\tblue"""`),
 });
 
 Deno.test({
+  name: "[TOML parser] multi-line basic string (CRLF)",
+  fn() {
+    const parse = ParserFactory(MultilineBasicString);
+    assertEquals(
+      parse(`"""\r
+Roses are red\r
+Violets are\\tblue"""`),
+      "Roses are red\r\nViolets are\tblue",
+    );
+    assertEquals(
+      parse(`"""\\\r
+    The quick brown \\\r
+    fox jumps over \\\r
+    the lazy dog.\\\r
+    """`),
+      "The quick brown fox jumps over the lazy dog.",
+    );
+  },
+});
+
+Deno.test({
   name: "[TOML parser] multi-line literal string",
   fn() {
     const parse = ParserFactory(MultilineLiteralString);
@@ -108,6 +129,19 @@ Deno.test({
 Roses are red
 Violets are\\tblue'''`),
       "Roses are red\nViolets are\\tblue",
+    );
+  },
+});
+
+Deno.test({
+  name: "[TOML parser] multi-line literal string (CRLF)",
+  fn() {
+    const parse = ParserFactory(MultilineLiteralString);
+    assertEquals(
+      parse(`'''\r
+Roses are red\r
+Violets are\\tblue'''`),
+      "Roses are red\r\nViolets are\\tblue",
     );
   },
 });

